_posts/en/newsletters/2020-04-22-newsletter.md

```diff
@@ -0,0 +1,185 @@
+  (pronounced "geeks"). The last remaining target platform for Guix reproducible
+  builds, macOS, has an open [draft PR][Bitcoin Core #17920].
+
+- [C-Lightning #3611][] adds a `keysend` plugin that allows a node to
+  securely receive [spontaneous payments][topic spontaneous
+  payments]---payments not preceded by the generation of an invoice.
+  These work by having the sender of a payment choose its payment
+  preimage (normally chosen by the receiver), derive its payment hash
+  (normally included in an invoice), encrypt the preimage to the
+  receiver's node pubkey, and send a payment with the encrypted data to
+  the receiver secured by the payment hash.  The receiver decrypts the
+  preimage and uses it to claim the payment like normal.  In order to
+  allow payment tracking, `lightningd` automatically creates an internal
+  invoice for the decrypted preimage before claiming a spontaneous
+  payment.  An obvious use for spontaneous payments is donations, but a
+  less obvious use is for sending chat messages with a payment such as
+  via the LND-compatible [WhatSat][] software and the
+  C-Lightning-compatible [noise][] plugin.
+
+- [C-Lightning #3623][] adds a minimal implementation (only available
+  with the configuration parameter `--enable-experimental-features`) for
+  spending payments using [blinded paths][].  As described in
+  [Newsletter #92][news92 blinded paths], blinded paths make it possible
+  to route a payment without the originator learning the destination's
+  network identity or the full path used.  This not only improves the
+  privacy of the origination and destination endpoints but also the
+  privacy of any unannounced nodes along the blinded paths.  The minimal
+  implementation in this PR is mainly designed for testing, such as with
+  a work-in-progress implementation being developed for Eclair.
+
+- [LND #4163][] adds a `version` RPC that returns information about the
+  LND server version and build flags.  This makes it easier for
+  applications to ensure they're compatible with the currently running
+  LND version.
+
+- [Rust-Lightning #441][] adds support for sending and receiving basic
+  [multipath payments][topic multipath payments] which allow a payment
+  to be split into multiple parts that may each be relayed through
+  different channels.  This implementation is not fully usable yet as
+  follow-up pull requests are needed to [add support for route
+  finding][rl mpp next] and for [timing out partial
+  payments][rust-lightning #587].
+
+## Footnotes
+
+    {% capture today-private %}Inputs:<br>&nbsp;&nbsp;Alice (2 BTC)<br>&nbsp;&nbsp;Alice (2 BTC)<br><br>Outputs:<br>&nbsp;&nbsp;Alice's change (1 BTC)<br>&nbsp;&nbsp;Bob's revenue (3 BTC){% endcapture %}
+    {% capture today-public %}Inputs:<br>&nbsp;&nbsp;Spender (2 BTC)<br>&nbsp;&nbsp;Spender (2 BTC)<br><br>Outputs:<br>&nbsp;&nbsp;Spender or Receiver (1 BTC)<br>&nbsp;&nbsp;Spender or Receiver (3 BTC){% endcapture %}
```

Comment:
I think an observer might be able to reasonably assume this. If the 1BTC was for the receiver, the sender would only use one 2BTC input.

Reply:
Good point.  What you're implying is known as the Unnecessary Input Heuristic (UIH); see the discussion at the end of [this gist](https://gist.github.com/AdamISZ/4551b947789d3216bacfcb7af25e029e).  Interesting. LaurentMT has done some block chain scanning and [found](https://gist.github.com/AdamISZ/4551b947789d3216bacfcb7af25e029e#gistcomment-2799709) a high percentage of transactions that violate UIH---meaning that assumptions in this case may not be anywhere near warranted.  However, I can tweak the numbers so that this conflating issue doesn't come up.